### ISSUE TITLE
JournalReader seek methods should return the next index

### ISSUE THREAD

**Issue by npepinpe** [zeebe-io/zeebe#6354]
**Description**

To simplify things for #6307, all `JournalReader#seek*` methods should return the index of the entry `#next()` would return after the seek is complete, or `Journal#getFirstIndex()` if the log is currently empty (debatable - should we return some known value like `-1` instead?). Callers are still expected to call `hasNext()` before `next()` anyway, so it shouldn't be a big problem if the log is empty.

Once the new serialization format is in place, we could now consider dropping this instead exposing `getCurrentEntry()` - I will write another issue for this.

Blocks #6307
